signingscript/test/integration/test_autograph.py

```diff
@@ -0,0 +1,224 @@
+  "extra": {},
+  "metadata": {
+    "description": "Markdown description of **what** this task does",
+    "name": "Example Task",
+    "owner": "name@example.com",
+    "source": "https://tools.taskcluster.net/task-creator/"
+  },
+  "payload": {
+    "upstreamArtifacts": [{
+      "taskId": "upstream-task-id1",
+      "taskType": "build",
+      "paths": [],      # Configured by test
+      "formats": []     # Configured by test
+    }],
+    "maxRunTime": 600
+  },
+  "priority": "normal",
+  "provisionerId": "test-dummy-provisioner",
+  "requires": "all-completed",
+  "retries": 0,
+  "routes": [],
+  "schedulerId": "-",
+  "scopes": [
+    "project:releng:signing:cert:dep-signing",
+    "project:releng:signing:autograph:dep-signing",
+    # Format added by test
+  ],
+  "tags": {},
+  "taskGroupId": "CRzxWtujTYa2hOs20evVCA",
+  "workerType": "dummy-worker-aki"
+}
+
+
+async def _download_file(url, abs_filename, chunk_size=128):
+    parent_dir = os.path.dirname(abs_filename)
+    async with aiohttp.ClientSession() as session:
+        async with session.get(url) as resp:
+            makedirs(parent_dir)
+            with open(abs_filename, "wb") as fd:
+                while True:
+                    chunk = await resp.content.read(chunk_size)
+                    if not chunk:
+                        break
+                    fd.write(chunk)
+
+
+async def _download_files(urls, work_dir):
+    urls_per_on_disk_path = {
+        os.path.join(work_dir, 'cot/upstream-task-id1/', os.path.basename(url)): url
+        for url in urls
+    }
+    download_tasks = [_download_file(url, on_disk_path) for on_disk_path, url in urls_per_on_disk_path.items()]
+    print('Downloading original files...')
+    await asyncio.gather(*download_tasks)
+    print('Downloaded original files: {}'.format(urls))
+    return urls_per_on_disk_path.keys()
+
+
+def _write_server_config(tmpdir):
+    server_config_path = os.path.join(tmpdir, 'server_config.json')
+    with open(server_config_path, mode='w') as f:
+        json.dump(DEFAULT_SERVER_CONFIG, f)
+
+    return server_config_path
+
+
+def _craft_task(file_names, signing_format):
+    task = copy.deepcopy(DEFAULT_TASK)
+    task['payload']['upstreamArtifacts'][0]['paths'] = file_names
+    task['payload']['upstreamArtifacts'][0]['formats'] = [signing_format]
+    task['scopes'].append('project:releng:signing:format:{}'.format(signing_format))
+
+    return task
+
+
+@pytest.mark.asyncio
+@skip_when_no_network
+async def test_integration_autograph_mar(context, tmpdir):
+    urls_to_download = (
+        'https://archive.mozilla.org/pub/firefox/nightly/2017/10/2017-10-02-22-02-04-mozilla-central/firefox-mozilla-central-58.0a1-linux-x86_64-en-US-20171001220301-20171002220204.partial.mar',
+        'https://archive.mozilla.org/pub/firefox/nightly/2017/10/2017-10-02-10-01-34-mozilla-central/firefox-mozilla-central-58.0a1-linux-x86_64-en-US-20171001220301-20171002100134.partial.mar',
```

Comment:
You all would have a better idea, but I think these might 404 at some point

Reply:
I think we should just stuff these in the repo (or maybe find some smaller MARs/APKs to include, to not bloat the repo too much).

Reply:
Done! MARs are 6+MB and APK is 3+MB.